src/main/java/net/minestom/server/event/server/ServerListPingEvent.java

```diff
@@ -0,0 +1,198 @@
+package net.minestom.server.event.server;
+
+import net.kyori.adventure.text.Component;
+import net.minestom.server.MinecraftServer;
+import net.minestom.server.entity.Player;
+import net.minestom.server.event.CancellableEvent;
+import net.minestom.server.event.Event;
+import net.minestom.server.network.player.NettyPlayerConnection;
+import net.minestom.server.network.player.PlayerConnection;
+import net.minestom.server.ping.ResponseData;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.UUID;
+
+public class ServerListPingEvent extends Event implements CancellableEvent {
```

Comment:
I believe that only #getResponseData should be used to modify the ping result, otherwise we end up with a lot of code duplication and be hard to maintain